dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/models/SubSnpCoreFields.java

```diff
@@ -279,11 +294,24 @@ public String getAllelesInForwardStrand() {
                 ^ this.getContigOrientation().equals(Orientation.FORWARD);
 
         String alleles = this.getAlleles();
-        if (forward) {
-            return alleles;
-        } else {
-            return calculateReverseComplement(alleles);
+        if (!forward) {
+            alleles = calculateReverseComplement(alleles);
+        }
+        String[] split = alleles.split("/", -1);
+        return Stream.of(split).map(this::getTrimmedAllele).collect(Collectors.joining("/"));
+    }
+
+    public String[] getSecondaryAlternatesInForwardStrand() {
+        String[] alleles = this.getAllelesInForwardStrand().split("/", -1);
+        List<String> secondaryAlternates = new ArrayList<>(Arrays.asList(alleles));
+        for (String allele : alleles) {
+            if (allele.equals(this.getReferenceInForwardStrand())
+                    || allele.equals(this.getAlternateInForwardStrand())) {
```

Comment:
We may need to keep an eye on performance after some runs. If we are going to execute these methods several times it could be worth running them always once and storing the value in an attribute.

Reply:
getSecondaryAlternatesInForwardStrand is just called once. However, getReferenceInForwardStrand and getAlternateInForwardStrand will be called at least two times each. Those are not super intensive methods, the only processing here is calculateReverseComplement, and it will be called just for the reverse strand alleles.

Also, getSecondaryAlternatesInForwardStrand can be a static method, it does not depend on any member of the SubSnpCoreFields class. A more object-oriented approach would be implement a "ReverseCalculator" class, that could be injected to SubSnpCoreFields, and this class could benefit of caching. 

Anyway, I think we can leave it like it is for now. Any thoughts @jmmut ?

Reply:
Agreed to measure performance before doing any improvement